### ISSUE TITLE
Bug: clip_grad_norm_ doesn't work with FSDP2

### ISSUE THREAD

**Issue by zzhhjjj** [pytorch/pytorch#170862]
### 🐛 Describe the bug

The current `clip_grad_norm_` does not seem to work correctly with FSDP2.
1. It returns the norm of partial (local) parameters rather than the norm over all parameters.
2. It appears to use an incorrect gradient norm when normalizing the gradients.

The correct approach should follow what is implemented in TorchTitan:

https://github.com/pytorch/pytorch/blob/d38164a545b4a4e4e0cf73ce67173f70574890b6/torch/nn/utils/clip_grad.py#L186-L233
https://github.com/pytorch/torchtitan/blob/658f94cb9b27963784268185ed75c524f3b9b2be/torchtitan/distributed/utils.py#L399-L477

### Versions

PyTorch version: 2.9.1+cu128
